javascripts/2048.js

```diff
@@ -1,32 +1,400 @@
+// Constants -----------------------------------------------------------
+const MINSTARTINGTILE = 2;
+const MAXSTARTINGTILE = 4;
+const MINBOARDLOCALE = 0; // starting array index
+const MAXBOARDLOCALE = 3; // highest array index
+const BOARDSIZE = 4; // anything less than 4 is valid
+const WINNINGTILE = 2048;
+// Global Variables ------------------------------------------------------
+var board;
+var score;
+var actionOccurred;
+var alreadyWon;
+var gameOver;
+
 $(document).ready(function() {
-  console.log('ready!');
-  $('body').keydown(function(event){
-    var arrow_keys = [37, 38, 39, 40];
-    if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
-      event.preventDefault();
+  begin();
+
+  $('#newgame').click(function() {
+    clearBoard();
+    begin();
+  });
+
+  function begin() {
+    board = []; // creates an empty board
+    for (i = 0; i < BOARDSIZE; i++) { // fills the board: creates a 2D array
+      board[i] = new Array(BOARDSIZE);
+    }
+
+    // creates two tiles to start with
+    createTile();
+    createTile();
+
+    score = 0; // sets score to 0 for a new game
+    alreadyWon = false;
+    gameOver = false;
+    changeDisplayedScore();
+  }
+
+  $('body').keydown(function(event) {
+    if (!gameOver) {
+      var arrowKeys = [37, 38, 39, 40];
```

Comment:
These numbers are magic values. Consider translating them into something more legible here, at your earliest opportunity. Strings like 'left' and 'up' would work fine.